task.ps1

```diff
@@ -21,3 +22,28 @@ $nsgRuleHTTP = New-AzNetworkSecurityRuleConfig -Name HTTP  -Protocol Tcp -Direct
 New-AzNetworkSecurityGroup -Name $networkSecurityGroupName -ResourceGroupName $resourceGroupName -Location $location -SecurityRules $nsgRuleSSH, $nsgRuleHTTP
 
 # ↓↓↓ Write your code here ↓↓↓
+Write-Host "Creating a virtual network $virtualNetworkName and a subnet $subnetName ..."
+New-AzVirtualNetwork -Name $virtualNetworkName -ResourceGroupName $resourceGroupName -Location $location -AddressPrefix $vnetAddressPrefix -Subnet @(New-AzVirtualNetworkSubnetConfig -Name $subnetName -AddressPrefix $subnetAddressPrefix)
+
+Write-Host "Creating a public IP address $publicIpAddressName with DNS label ..."
+New-AzPublicIpAddress -ResourceGroupName $resourceGroupName -Location $location -Name $publicIpAddressName -AllocationMethod Static -Sku Basic -DomainNameLabel $domainNameLabel
+
+Write-Host "Creating a SSH key resource $sshKeyName ..."
+New-AzSshKey -ResourceGroupName $resourceGroupName -Name $sshKeyName -PublicKey "$sshKeyPublicKey"
+
+Write-Host "Creating a virtual machine $vmName ..."
+$credential = Get-Credential
+
+New-AzVm `
+    -ResourceGroupName $resourceGroupName `
+    -Name $vmName `
+    -Location $location `
+    -image $vmImage `
+    -size $vmSize `
+    -VirtualNetworkName $virtualNetworkName `
+    -SubnetName $subnetName `
+    -SecurityGroupName $networkSecurityGroupName `
+    -PublicIpAddressName $publicIpAddressName `
+    -OpenPorts 22,8080 `
+    -SshKeyName $sshKeyName `
```

Comment:
Ensure that the SSH key resource `$sshKeyName` is correctly associated with the virtual machine. Verify that the SSH key is correctly configured and accessible.